tiledb/query_condition.py

```diff
@@ -10,8 +12,23 @@
 filtering query results on attribute values.
 """
 
+QueryConditionNodeElem = TypeVar(
```

Comment:
IIUC this should be a `Union` rather than a `TypeVar`, because the in/out types can be different.